### PR TITLE
Add elvis; fixes #141

### LINKED ISSUE #141: Add elvis
It would be nice to add https://github.com/inaka/elvis to CI pipeline and fix style issues it reports (or tune some of those rules if they are too strict).

For example, on current devel with default `elvis.config` it's output is:

```
# src/epgsql.erl [FAIL]
  - variable_naming_convention
    - The variable "Typed_Parameters" on line 227 does not respect the format defined by the regular expression '"^_?([A-Z][0-9a-zA-Z]*)$"'.
    - The variable "Typed_Parameters" on line 228 does not respect the format defined by the regular expression '"^_?([A-Z][0-9a-zA-Z]*)$"'.
    - The variable "Typed_Parameters" on line 237 does not respect the format defined by the regular expression '"^_?([A-Z][0-9a-zA-Z]*)$"'.
    - The variable "Typed_Parameters" on line 238 does not respect the format defined by the regular expression '"^_?([A-Z][0-9a-zA-Z]*)$"'.
    - The variable "Typed_Parameters" on line 247 does not respect the format defined by the regular expression '"^_?([A-Z][0-9a-zA-Z]*)$"'.
    - The variable "Typed_Parameters" on line 248 does not respect the format defined by the regular expression '"^_?([A-Z][0-9a-zA-Z]*)$"'.
  - no_spec_with_records
    - The spec in line 262 uses a record, please define a type for the record and use that instead.
    - The spec in line 272 uses a record, please define a type for the record and use that instead.
    - The spec in line 287 uses a record, please define a type for the record and use that instead.
    - The spec in line 296 uses a record, please define a type for the record and use that instead.
  - god_modules
    - This module has too many functions (37). Consider breaking it into a number of modules.
  - operator_spaces
    - Missing space right "," on line 175
  - no_tabs
    - Line 130 has a tab at column 0.
    - Line 131 has a tab at column 0.
    - Line 132 has a tab at column 0.
    - Line 133 has a tab at column 0.
  - line_length
    - Line 110 is too long:     {ok, ColumnsDescription :: [#column{}], RowsValues :: [RowType]} |                            % select.
    - Line 111 is too long:     {ok, Count :: non_neg_integer()} |                                                            % update/insert/delete.
    - Line 112 is too long:     {ok, Count :: non_neg_integer(), ColumnsDescription :: [#column{}], RowsValues :: [RowType]}. % update/insert/delete + returning.
    - Line 243 is too long: -spec prepared_query(C::connection(), Name::string(), Parameters::[bind_param()]) -> reply(equery_row())..
    - Line 350 is too long: %% @doc sends last flushed and applied WAL positions to the server in a standby status update message via given `Connection'.
    - Line 354 is too long: -spec start_replication(connection(), string(), Callback, cb_state(), string(), string()) -> ok | error_reply() when.
    - Line 360 is too long: %% `Callback'        - Callback module which should have the callback functions implemented for message processing..
    - Line 369 is too long:     gen_server:call(Connection, {start_replication, ReplicationSlot, Callback, CbInitState, WALPosition, PluginOpts})..
# src/epgsql_binary.erl [FAIL]
  - operator_spaces
    - Missing space right "," on line 79
    - Missing space right "," on line 79
    - Missing space right "," on line 143
  - macro_module_names
    - Don't use macros (like datetime on line 64) as module names.
    - Don't use macros (like datetime on line 65) as module names.
    - Don't use macros (like datetime on line 66) as module names.
    - Don't use macros (like datetime on line 67) as module names.
    - Don't use macros (like datetime on line 68) as module names.
    - Don't use macros (like datetime on line 69) as module names.
    - Don't use macros (like datetime on line 98) as module names.
    - Don't use macros (like datetime on line 99) as module names.
    - Don't use macros (like datetime on line 100) as module names.
    - Don't use macros (like datetime on line 101) as module names.
    - Don't use macros (like datetime on line 102) as module names.
    - Don't use macros (like datetime on line 103) as module names.
  - macro_names
    - Invalid macro name datetime on line 17. Use UPPER_CASE.
  - line_length
    - Line 74 is too long: encode(jsonb, B, _) when is_binary(B)       -> <<(byte_size(B) + 1):?int32, ?JSONB_VERSION_1:8, B/binary>>;.
    - Line 76 is too long: encode({array, char}, L, Codec) when is_list(L) -> encode_array(bpchar, type2oid(bpchar, Codec), L, Codec);.
    - Line 77 is too long: encode({array, Type}, L, Codec) when is_list(L) -> encode_array(Type, type2oid(Type, Codec), L, Codec);.
    - Line 221 is too long: decode_hstore1(N, <<KeyLen:?int32, Key:KeyLen/binary, ValLen:?int32, Value:ValLen/binary, Rest/binary>>, Acc) ->.
    - Line 277 is too long: decode_int4range(<<2:1/big-signed-unit:8, 4:?int32, From:?int32, 4:?int32, To:?int32>>) -> {From, To};.
    - Line 283 is too long: decode_int8range(<<2:1/big-signed-unit:8, 8:?int32, From:?int64, 8:?int32, To:?int64>>) -> {From, To};.
# src/epgsql_fdatetime.erl [FAIL]
  - macro_names
    - Invalid macro name postgres_epoc_jdate on line 9. Use UPPER_CASE.
    - Invalid macro name postgres_epoc_secs on line 10. Use UPPER_CASE.
    - Invalid macro name mins_per_hour on line 12. Use UPPER_CASE.
    - Invalid macro name secs_per_day on line 13. Use UPPER_CASE.
    - Invalid macro name secs_per_hour on line 14. Use UPPER_CASE.
    - Invalid macro name secs_per_minute on line 15. Use UPPER_CASE.
# src/epgsql_idatetime.erl [FAIL]
  - macro_names
    - Invalid macro name postgres_epoc_jdate on line 9. Use UPPER_CASE.
    - Invalid macro name postgres_epoc_usecs on line 10. Use UPPER_CASE.
    - Invalid macro name mins_per_hour on line 12. Use UPPER_CASE.
    - Invalid macro name secs_per_minute on line 13. Use UPPER_CASE.
    - Invalid macro name usecs_per_day on line 15. Use UPPER_CASE.
    - Invalid macro name usecs_per_hour on line 16. Use UPPER_CASE.
    - Invalid macro name usecs_per_minute on line 17. Use UPPER_CASE.
    - Invalid macro name usecs_per_sec on line 18. Use UPPER_CASE.
# src/epgsql_sock.erl [FAIL]
  - variable_naming_convention
    - The variable "Typed_Parameters" on line 293 does not respect the format defined by the regular expression '"^_?([A-Z][0-9a-zA-Z]*)$"'.
    - The variable "Typed_Parameters" on line 294 does not respect the format defined by the regular expression '"^_?([A-Z][0-9a-zA-Z]*)$"'.
    - The variable "Typed_Parameters" on line 316 does not respect the format defined by the regular expression '"^_?([A-Z][0-9a-zA-Z]*)$"'.
    - The variable "Typed_Parameters" on line 317 does not respect the format defined by the regular expression '"^_?([A-Z][0-9a-zA-Z]*)$"'.
  - dont_repeat_yourself
    - The code in the following (LINE, COL) locations has the same structure: (264, 9), (276, 9).
    - The code in the following (LINE, COL) locations has the same structure: (265, 9), (277, 9).
  - state_record_and_type
    - This module implements an OTP behavior and has a 'state' record but is missing a 'state()' type.
  - invalid_dynamic_call
    - Remove the dynamic function call on line 423. Only modules that define callbacks should make dynamic calls.
    - Remove the dynamic function call on line 858. Only modules that define callbacks should make dynamic calls.
  - no_if_expression
    - Replace the 'if' expression on line 517 with a 'case' expression or function clauses.
  - operator_spaces
    - Missing space right "," on line 419
  - line_length
    - Line 140 is too long:     send(State, ?COPY_DATA, epgsql_wire:encode_standby_status_update(ReceivedLSN, FlushedLSN, AppliedLSN)),.
    - Line 141 is too long:     {reply, ok, State#state{repl_last_flushed_lsn = FlushedLSN, repl_last_applied_lsn = AppliedLSN}};.
    - Line 356 is too long: command({start_replication, ReplicationSlot, Callback, CbInitState, WALPosition, PluginOpts}, State) ->.
    - Line 526 is too long:         {_, {Command, #statement{columns = C}, _}}  when Command == equery; Command == prepared_query ->.
    - Line 833 is too long: on_message({?COPY_DATA, _Data}, #state{repl_cbmodule = undefined, repl_receiver = undefined} = State) ->.
    - Line 837 is too long: on_message({?COPY_DATA, <<?PRIMARY_KEEPALIVE_MESSAGE:8, LSN:?int64, _Timestamp:?int64, ReplyRequired:8>>},.
    - Line 838 is too long:     #state{repl_last_flushed_lsn = LastFlushedLSN, repl_last_applied_lsn = LastAppliedLSN} = State) ->.
    - Line 849 is too long: on_message({?COPY_DATA, <<?X_LOG_DATA, StartLSN:?int64, EndLSN:?int64, _Timestamp:?int64, WALRecord/binary>>},.
    - Line 855 is too long: on_message({?COPY_DATA, <<?X_LOG_DATA, StartLSN:?int64, EndLSN:?int64, _Timestamp:?int64, WALRecord/binary>>},.
# src/epgsql_wire.erl [FAIL]
  - variable_naming_convention
    - The variable "_Table_Oid" on line 149 does not respect the format defined by the regular expression '"^_?([A-Z][0-9a-zA-Z]*)$"'.
    - The variable "_Attrib_Num" on line 149 does not respect the format defined by the regular expression '"^_?([A-Z][0-9a-zA-Z]*)$"'.
    - The variable "Type_Oid" on line 149 does not respect the format defined by the regular expression '"^_?([A-Z][0-9a-zA-Z]*)$"'.
    - The variable "Type_Oid" on line 153 does not respect the format defined by the regular expression '"^_?([A-Z][0-9a-zA-Z]*)$"'.
  - line_length
    - Line 239 is too long:     Timestamp = ((MegaSecs * 1000000 + Secs) * 1000000 + MicroSecs) - 946684800*1000000, %% microseconds since midnight on 2000-01-01.
# src/epgsqla.erl [FAIL]
  - no_spec_with_records
    - The spec in line 79 uses a record, please define a type for the record and use that instead.
    - The spec in line 83 uses a record, please define a type for the record and use that instead.
    - The spec in line 100 uses a record, please define a type for the record and use that instead.
    - The spec in line 110 uses a record, please define a type for the record and use that instead.
    - The spec in line 114 uses a record, please define a type for the record and use that instead.
  - god_modules
    - This module has too many functions (30). Consider breaking it into a number of modules.
# src/epgsqli.erl [FAIL]
  - no_spec_with_records
    - The spec in line 78 uses a record, please define a type for the record and use that instead.
    - The spec in line 82 uses a record, please define a type for the record and use that instead.
    - The spec in line 99 uses a record, please define a type for the record and use that instead.
    - The spec in line 109 uses a record, please define a type for the record and use that instead.
    - The spec in line 113 uses a record, please define a type for the record and use that instead.
  - god_modules
    - This module has too many functions (29). Consider breaking it into a number of modules.
  - line_length
    - Line 50 is too long:     epgsqla:complete_connect(C, incremental(C, {connect, Host, Username, Password, epgsql:to_proplist(Opts)}))..
# src/ewkb.erl [FAIL]
  - operator_spaces
    - Missing space right "," on line 188
    - Missing space right "," on line 190
    - Missing space right "," on line 190
    - Missing space right "," on line 192
    - Missing space right "," on line 192
    - Missing space right "," on line 194
    - Missing space right "," on line 194
    - Missing space right "," on line 194
    - Missing space right "," on line 206
    - Missing space right "," on line 207
    - Missing space right "," on line 208
    - Missing space right "," on line 209
    - Missing space right "," on line 210
    - Missing space right "," on line 211
    - Missing space right "," on line 212
    - Missing space right "," on line 213
    - Missing space right "," on line 214
    - Missing space right "," on line 215
    - Missing space right "," on line 216
    - Missing space right "," on line 217
    - Missing space right "," on line 218
    - Missing space right "," on line 219
    - Missing space right "," on line 220
    - Missing space right "," on line 221
    - Missing space right "," on line 222
    - Missing space right "," on line 223
    - Missing space right "," on line 249
    - Missing space right "," on line 257
    - Missing space right "," on line 258
    - Missing space right "," on line 259
    - Missing space right "," on line 260
  - no_trailing_whitespace
    - Line 63 has 1 trailing whitespace characters.
    - Line 98 has 2 trailing whitespace characters.
    - Line 99 has 2 trailing whitespace characters.
    - Line 120 has 1 trailing whitespace characters.
    - Line 121 has 1 trailing whitespace characters.
    - Line 132 has 1 trailing whitespace characters.
```

### PR DESCRIPTION
**Pull request by seriyps**
Uses rebar3 lint plugin. Fixes #141

### CODE REVIEW

File: src/datatypes/epgsql_codec_datetime.erl [hunk 1]
```diff
@@ -46,10 +46,14 @@ names() ->
 
 %% FIXME: move common logick out from fdatetime/idatetime; make them more
 %% low-level
-encode(Val, Type, Mod) ->
-    Mod:encode(Type, Val).
-
-decode(Bin, Type, Mod) ->
-    Mod:decode(Type, Bin).
```
Comment (seriyps) [inline src/datatypes/epgsql_codec_datetime.erl]: This was https://github.com/inaka/elvis_core/wiki/Rules#invalid-dynamic-calls

File: src/epgsql_fdatetime.erl [hunk 2]
```diff
@@ -6,22 +6,22 @@
 
 -include("protocol.hrl").
 
--define(postgres_epoc_jdate, 2451545).
--define(postgres_epoc_secs, 946684800).
+-define(POSTGRES_EPOC_JDATE, 2451545).
+-define(POSTGRES_EPOC_SECS, 946684800).
```
Comment (seriyps) [inline src/epgsql_fdatetime.erl]: https://github.com/inaka/elvis_core/wiki/Rules#macro-names

File: src/epgsql_sock.erl [hunk 3]
```diff
@@ -366,9 +366,8 @@ do_send(gen_tcp, Sock, Bin) ->
         error:_Error ->
             {error, einval}
     end;
-
-do_send(Mod, Sock, Bin) ->
-    Mod:send(Sock, Bin).
+do_send(ssl, Sock, Bin) ->
+    ssl:send(Sock, Bin).
```
Comment (seriyps) [inline src/epgsql_sock.erl]: Again https://github.com/inaka/elvis_core/wiki/Rules#invalid-dynamic-calls

File: src/datatypes/epgsql_codec_datetime.erl (hunk 1 above)
Comment (davidw) [reply src/datatypes/epgsql_codec_datetime.erl]: logick should be logic

**Comment by seriyps** [comment]
Thanks! Typo fixed, merged

**Pull request merged**